### PR TITLE
organize docs

### PR DESCRIPTION
**Pull request by ramineni**
**What this PR does / why we need it**:
This PR organizes docs to respective folders for better readability.

**Which issue this PR fixes(if applicable)**:
fixes #

**Special notes for reviewers**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: docs/keystone-auth/using-auth-data-synchronization.md [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 Authentication synchronization is an experimental feature allowing to create Kubernetes auth related entities during user authentication.
 
-As an example, you can take the synchronization of Keystone projects with Kubernetes namespaces. If the user belongs to the project in Keystone, then, when attempting to authenticate in Kubernetes using [k8s-keystone-auth](./using-keystone-webhook-authenticator-and-authorizer.md) webhook for the first time, a Kubernetes namespace will automatically be created for the user.
+As an example, you can take the synchronization of Keystone projects with Kubernetes namespaces. If the user belongs to the project in Keystone, then, when attempting to authenticate in Kubernetes using [k8s-keystone-auth](./keystone-auth/using-keystone-webhook-authenticator-and-authorizer.md) webhook for the first time, a Kubernetes namespace will automatically be created for the user.
```
Comment (lingxiankong) [inline docs/keystone-auth/using-auth-data-synchronization.md:19]: Should it be `../keystone-auth/using-keystone-webhook-authenticator-and-authorizer.md`?

**Review by lingxiankong — CHANGES_REQUESTED** [review]
Please also fix the folder name `barbical-kms-plugin`

File: docs/keystone-auth/using-auth-data-synchronization.md (hunk 1 above)
Comment (ramineni) [reply docs/keystone-auth/using-auth-data-synchronization.md]: @lingxiankong updated with ./using-keystone-webhook-authenticator-and-authorizer.md as they are in same folder.

**Comment by ramineni** [comment]
> Please also fix the folder name `barbical-kms-plugin`

good catch. Updated

**Comment by jichenjc** [comment]
looks good

I knew some tools might be helpful to check links in other repo, but don't remember the exact tool and name, do you happen to know so we can adopt later?

**Pull request merged**